### PR TITLE
Moving retry logic for UpgradeApplication to corresponding helper fun…

### PR DESCRIPTION
**Pull request by kuvinodms**
…ction

### CODE REVIEW

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1 [hunk 1]
```diff
@@ -240,7 +240,16 @@ function Get-ServiceFabricApplicationUpgradeAction
     )
 
     $global:operationId = $SF_Operations.GetApplicationUpgradeStatus
-    return Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName
+
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: This should be a separate function Wait-ServiceFabricApplicationUpgrade. get should just get the status and not wait for it to finish

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1 [hunk 2]
```diff
@@ -243,6 +243,26 @@ function Get-ServiceFabricApplicationUpgradeAction
...
+    $global:operationId = $SF_Operations.WaitApplicationUpgradeStatus
+
+    Write-Host (Get-VstsLocString -Key SFSDK_WaitingForUpgrade)
+    $upgradeStatusFetcher = { Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName }
+    $upgradeRetryEvaluator = { param($upgradeStatus) return ($upgradeStatus.UpgradeState -ne "RollingBackCompleted" -and $upgradeStatus.UpgradeState -ne "RollingForwardCompleted") }
+    return Invoke-ActionWithRetries -Action $upgradeStatusFetcher `
+        -ResultRetryEvaluator $upgradeRetryEvaluator `
+        -MaxTries 2147483647 `
+        -RetryIntervalInSeconds 3 `
+        -RetryableExceptions @("System.Fabric.FabricTransientException") `
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: Should we add TimeoutException here as well.

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1 [hunk 3]
```diff
@@ -243,6 +243,26 @@ function Get-ServiceFabricApplicationUpgradeAction
     return Get-ServiceFabricApplicationUpgrade -ApplicationName $ApplicationName
 }
 
+function Wait-ServiceFabricApplicationUpgradeAction
```
Comment (bishal-pdMSFT) [inline Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: Please think about adding L0 test for this function.

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1 (hunk 2 above)
Comment (kuvinodms) [reply Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: Adding timeout exception is not a harm. Adding!

File: Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1 (hunk 3 above)
Comment (kuvinodms) [reply Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: Its a minor work but I don't want to spend any more time over this. Either I'll add it later or you take it up.

Comment (bishal-pdMSFT) [reply Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: You own this PR and hence you should decide whether to add test or not.

Comment (kuvinodms) [reply Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: Please approve if there are not any further concerns.

Comment (bishal-pdMSFT) [reply Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: If you are not adding L0, open a user story/task to track that.

Comment (kuvinodms) [reply Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1]: Here you go https://mseng.visualstudio.com/DefaultCollection/VSOnline/_workitems/edit/1296454

**Pull request merged**